src/client/dfuse/ops/read.c

```diff
@@ -51,6 +51,55 @@ dfuse_cb_read_complete(struct dfuse_event *ev)
 	d_slab_release(ev->de_eqt->de_read_slab, ev);
 }
 
+static bool
+dfuse_readahead_reply(fuse_req_t req, size_t len, off_t position, struct dfuse_obj_hdl *oh)
+{
+	size_t reply_len;
+
+	if (oh->doh_readahead->dra_rc) {
+		DFUSE_REPLY_ERR_RAW(oh, req, oh->doh_readahead->dra_rc);
+		return true;
+	}
+
+	if (!oh->doh_linear_read || oh->doh_readahead->dra_ev == NULL) {
+		DFUSE_TRA_DEBUG(oh, "Readahead disabled");
+		return false;
+	}
+
+	if (oh->doh_linear_read_pos != position) {
+		DFUSE_TRA_DEBUG(oh, "disabling readahead");
```

Comment:
So "linear read" is synonymous with "readahead"?

Reply:
Apparently yes.  This PR started are read-ahead after the first read before I realised that the speculative read could happen on open.  This is debugging so I think it's Ok though.

Reply:
I was thinking from the perspective of the variable names, etc., but not asking you to change this

Reply:
https://github.com/daos-stack/daos/blob/master/src/client/dfuse/dfuse.h#L117-L126

`linear read` means the file was opened, read from start to end without seeking and closed.  We already track this to short-circuit the last zero-length read so we use that data to know if to use the pre-read buffer.